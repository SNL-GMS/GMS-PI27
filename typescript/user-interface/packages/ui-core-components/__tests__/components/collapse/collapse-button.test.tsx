import { fireEvent, render } from '@testing-library/react';
import * as React from 'react';

import { CollapseButton } from '../../../src/ts/components/collapse/collapse-button';
import type * as CollapseButtonTypes from '../../../src/ts/components/collapse/types';

const props: CollapseButtonTypes.CollapseButtonProps = {
  buttonText: 'collapse-button',
  isLoading: false,
  isCollapsed: false,
  onClick: jest.fn()
};

describe('CollapseButton', () => {
  it('to be defined', () => {
    expect(CollapseButton).toBeDefined();
  });

  it('CollapseButton renders', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const { container } = render(<CollapseButton {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('CollapseButton functions and clicks', () => {
    const mockOnClick = jest.fn();

    const result = render(
      <CollapseButton
        buttonText="collapse-button"
        isLoading={false}
        isCollapsed
        onClick={mockOnClick}
      />
    );

    const button = result.container.querySelector(`Button`);
    if (button) {
      fireEvent.click(button);
    }
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
