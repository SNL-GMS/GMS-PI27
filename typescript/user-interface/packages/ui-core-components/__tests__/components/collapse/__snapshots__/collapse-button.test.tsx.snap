// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapseButton CollapseButton renders 1`] = `
<div>
  <button
    class="bp5-button collapse-button--closed button--full-width"
    type="button"
  >
    <span
      class="bp5-button-text"
    >
      collapse-button
    </span>
    <span
      aria-hidden="true"
      class="bp5-icon bp5-icon-standard bp5-icon-caret-down"
      data-icon="caret-down"
    />
  </button>
  <div
    class="bp5-collapse collapse-button__target--closed button--full-width"
  >
    <div
      aria-hidden="true"
      class="bp5-collapse-body"
      style="transform: translateY(-undefinedpx);"
    />
  </div>
</div>
`;
